registry/api/protocol/eureka/instance_handler.go

```diff
@@ -183,9 +166,18 @@ func (routes *Routes) registerInstance(w rest.ResponseWriter, r *rest.Request) {
 	w.WriteHeader(http.StatusNoContent)
 }
 
-func validateJSON(jsonString json.RawMessage) bool {
-	var js interface{}
-	return json.Unmarshal(jsonString, &js) == nil
+func extractTagsFromMetadata(jsonString json.RawMessage) ([]string, error) {
+	var md map[string]interface{}
+	if err := json.Unmarshal(jsonString, &md); err != nil {
+		return nil, err
+	}
+
+	var tags []string
+	if val, ok := md[metadataTags]; ok {
+		sval := fmt.Sprintf("%v", val)
+		tags = strings.Split(sval, metadataTagsDelimiter)
```

Comment:
Weird, I also wrote a comment about the delimiter, but apparently it disappeared from the review...
I was looking at [Spring Cloud](http://cloud.spring.io/spring-cloud-netflix/spring-cloud-netflix.html), and saw that `,` is normally used as a delimiter (search for "comma-separated"). However, if that clashes with Archaius, than we can just use `;`.